### PR TITLE
[#106] Display only one leaderboard table based on filter

### LINKED ISSUE #106: Two tables displayed when filtering leaderboard
When users apply a filter to the leaderboard, two tables are displayed. One table shows the filtered results while the other table displays the original, unfiltered results. Users have provided feedback that having two tables is confusing and affects their ability to quickly understand the information.

### PR DESCRIPTION
**Pull request by kangsuhyun-yanolja**
This change updates the leaderboard display behavior to show only one table at a time.

- If the language filter is set to 'Any', the original, unfiltered leaderboard is displayed.
- If a specific language is selected, a filtered leaderboard is shown.

### Before Change
스크린샷 2024-06-05 오후 12 40 41

### After Change
스크린샷 2024-06-05 오후 12 41 44

resolved #106

### CODE REVIEW

File: leaderboard.py [hunk 1]
```diff
@@ -20,7 +21,7 @@
   firebase_admin.initialize_app(credentials.Certificate(get_credentials_json()))
   db = firestore.client()
 
-SUPPORTED_TRANSLATION_LANGUAGES = [
+SUPPORTED_LANGUAGES = [
```
Comment (geon-yanolja) [inline leaderboard.py:24]: Is it going to deal with translation only? so it's clear the supported languages will be only about translation?

wondering why you removed `translation`.

**Comment by geon-yanolja** [comment]
how can i see the website? or one in dev environment if there is one?

**Comment by kangsuhyun-yanolja** [comment]
I just updated the dev one. you can check it on Arena dev

File: leaderboard.py (hunk 1 above)
Comment (kangsuhyun-yanolja) [reply leaderboard.py:24]: no it's used in summarization too. so it's for the entire app.

**Comment by geon-yanolja** [comment]
Can i get the link?

**Comment by kangsuhyun-yanolja** [comment]
@rifqiyan dev is now working
@geon-yanolja I'll DM you

File: leaderboard.py (hunk 1 above)
Comment (geon-yanolja) [reply leaderboard.py:24]: will they always support the same languages? if so I think it'll be fine.

**Comment by geon-yanolja** [comment]
could it have the screenshot for before/after. Hard to tell what the original leaderboard and filtered one would look like.

File: leaderboard.py (hunk 1 above)
Comment (kangsuhyun-yanolja) [reply leaderboard.py:24]: yep

**Comment by kangsuhyun-yanolja** [comment]
@geon-yanolja yes but maybe checking https://huggingface.co/spaces/yanolja/arena would make it easier for you to compare!

File: leaderboard.py [hunk 2]
```diff
@@ -125,94 +120,94 @@ def update_filtered_leaderboard(tab, summary_lang: str, source_lang: str,
 
 def build_leaderboard():
   with gr.Tabs():
+
+    # Returns (original leaderboard, filtered leaderboard).
+    def toggle_leaderboard(value: str) -> Tuple[gr.Dataframe, gr.Dataframe]:
```
Comment (geon-yanolja) [inline leaderboard.py:125]: maybe `toggle_leaderboard(language: str) -> ...`?

File: leaderboard.py [hunk 3]
```diff
@@ -125,94 +120,94 @@ def update_filtered_leaderboard(tab, summary_lang: str, source_lang: str,
...
+          fn=update_filtered_leaderboard,
+          inputs=[
+              gr.State(LeaderboardTab.SUMMARIZATION), summary_language,
+              gr.State(),
+              gr.State()
+          ],
+          outputs=filtered_summarization).then(
+              fn=toggle_leaderboard,
+              inputs=summary_language,
+              outputs=[original_summarization, filtered_summarization])
```
Comment (rifqiyan) [inline leaderboard.py:163]: I checked `toggle_leaderboard` and it seems it create a new Dataframe with only `visible` field?
Does `gradio` automatically merge the value with `original_summarization` and `filtered_summarization`?

File: leaderboard.py [hunk 4]
```diff
@@ -125,94 +120,94 @@ def update_filtered_leaderboard(tab, summary_lang: str, source_lang: str,
...
+          value=lambda: load_elo_ratings(LeaderboardTab.SUMMARIZATION),
+          every=LEADERBOARD_UPDATE_INTERVAL,
+          elem_classes="leaderboard")
       gr.Markdown(LEADERBOARD_INFO)
 
+      summary_language.change(
+          fn=update_filtered_leaderboard,
+          inputs=[
+              gr.State(LeaderboardTab.SUMMARIZATION), summary_language,
+              gr.State(),
```
Comment (geon-yanolja) [inline leaderboard.py:157]: it seems like it's src, target language, but it's summarization, so it should be empty and `gr.State()` does that?

i'm not sure but could you make that part clear? or maybe split the functions for each... etc

File: leaderboard.py (hunk 3 above)
Comment (kangsuhyun-yanolja) [reply leaderboard.py:163]: `toggle_leaderboard` only updates visibility of existing original_summarization dataframe and filtered_summarization dataframe

File: leaderboard.py (hunk 4 above)
Comment (kangsuhyun-yanolja) [reply leaderboard.py:157]: yes. I needed to pass Gradio Component as inputs so I passed the empty State Component. I'll update it to gr.State(None)

**Review by rifqiyan — APPROVED** [review]

**Pull request merged**